sdk/tests/conformance/extensions/webgl-webcodecs-video-frame.html

```diff
@@ -0,0 +1,220 @@
+      var query = window.location.search.substring(1);
+      var vars = query.split("&");
+      for (var i = 0; i < vars.length; i++) {
+        var pair = vars[i].split("=");
+        if (pair[0] == variable) { return pair[1]; }
+      }
+      return false;
+    }
+
+    let th = parseInt(getQueryVariable('threshold'));
+    if (!isNaN(th))
+      pixelCompareTolerance = th;
+
+    async function startDrawing() {
+      let cnv = document.getElementById("src");
+      var ctx = cnv.getContext('2d', { alpha: false });
+
+      ctx.fillStyle = src_color;
+      let drawOneFrame = function (time) {
+        ctx.fillStyle = src_color;
+        ctx.fillRect(0, 0, src_width, src_height);
+        window.requestAnimationFrame(drawOneFrame);
+      }
+      window.requestAnimationFrame(drawOneFrame);
+    }
+
+    function captureAndEncode(processChunk) {
+      let cnv = document.getElementById("src");
+      let fps = 60;
+      let pending_outputs = 0;
+      let stream = cnv.captureStream(fps);
+      let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
+
+      const init = {
+        output: (chunk) => {
+          testPassed("Encode frame successfully.");
+          pending_outputs--;
+          processChunk(chunk);
+        },
+        error: (e) => {
+          testFailed("Failed to encode frame.");
+          finishTest();
+          vtr.stop();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        width: cnv.width,
+        height: cnv.height,
+        bitrate: 10e6,
+        framerate: fps,
+      };
+
+      let encoder = new VideoEncoder(init);
+      encoder.configure(config);
+
+      vtr.start((frame) => {
+        if (pending_outputs > 30) {
+          console.log("drop this frame");
+          // Too many frames in flight, encoder is overwhelmed
+          // let's drop this frame.
+          return;
+        }
+        if (frame_counter == kMaxFrame) {
+          return;
+        }
+
+        frame_counter++;
+        pending_outputs++;
+        const insert_keyframe = (frame_counter % 150) == 0;
+        encoder.encode(frame, { keyFrame: insert_keyframe });
+      });
+    }
+
+    function startDecodingAndRendering(cnv, handleFrame) {
+      const init = {
+        output: handleFrame,
+        error: (e) => {
+          testFailed("Failed to decode frame.");
+          finishTest();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        codedWidth: cnv.width,
+        codedHeight: cnv.height,
+        acceleration: "deny",
+
+      };
+
+      let decoder = new VideoDecoder(init);
+      decoder.configure(config);
+      return decoder;
+    }
+
+    // Compare pixels in 4 corner (upper left, upper right, bottom left, bottem right) and center
+    // with given color to decide whether the frame is matched.
+    function isFramePixelMatched(gl, th_per_pixel = pixelCompareTolerance) {
+      let pixels = new Uint8Array(src_width * src_height * 4);

```

Comment:
Instead of handwriting this comparison routine, could you please call WebGLTestUtils.checkCanvasRect a few times with different (x, y) coordinates and (width, height) of (1, 1), and with an appropriate tolerance?

Reply:
Fixed.